src/averageAges.js

```diff
@@ -37,7 +42,12 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter(person => person.sex === 'f'
+    && people.some(child => (person.name === child.mother)))
```

Comment:
[checklist point 2](https://github.com/mate-academy/js_average-ages/blob/master/checklist.md)

```suggestion
    ? people.filter(person => {
      const isMother = people.some(child => (person.name === child.mother));
      
      return person.sex === 'f' && isMother;
    })
```